python/taichi/lang/matrix.py

```diff
@@ -1045,13 +1045,488 @@ def identity(dt, n):
         return Matrix([[ops_mod.cast(int(i == j), dt) for j in range(n)]
                        for i in range(n)])
 
+    @staticmethod
+    def translate(dx, dy, dz):
+        """Constructs a translate Matrix with shape (4, 4).
+
+        Args:
+            dx (float): delta x.
+            dy (float): delta y.
+            dz (float): delta z.
+
+        Returns:
+            :class:`~taichi.Matrix`: An `4 x 4` translate matrix.
+
+        Example:
+
+            >>> import math
+            >>> ti.Matrix.translate(1, 2, 3)
+            [[ 1 0 0 1]
+             [ 0 1 0 2]
+             [ 0 0 1 3]
+             [ 0 0 0 1]]
+        """
+        return Matrix([[1., 0., 0., dx], [0., 1., 0., dy], [0., 0., 1., dz],
+                       [0., 0., 0., 1.]])
+
+    @staticmethod
+    def scale(sx, sy, sz):
+        """Constructs a scale Matrix with shape (4, 4).
+
+        Args:
+            sx (float): scale x.
+            sy (float): scale y.
+            sz (float): scale z.
+
+        Returns:
+            :class:`~taichi.Matrix`: An `4 x 4` translate matrix.
+
+        Example:
+
+            >>> import math
+            >>> ti.Matrix.scale(1, 2, 3)
+            [[ 1 0 0 0]
+             [ 0 2 0 0]
+             [ 0 0 3 0]
+             [ 0 0 0 1]]
+        """
+        return Matrix([[sx, 0., 0., 0.], [0., sy, 0., 0.], [0., 0., sz, 0.],
+                       [0., 0., 0., 1.]])
+
+    @staticmethod
+    def eulerAngleX(angleX):
+        """Creates a 3D 4 * 4 homogeneous rotation matrix from an euler angle X.
+
+        Args:
+            angleX (float): angle in radians unit around X axis
+
+        """
+        cosX = ops_mod.cos(angleX)
+        sinX = ops_mod.sin(angleX)
+
+        return Matrix([[1., 0., 0., 0.], [0., cosX, sinX, 0.],
+                       [0., -sinX, cosX, 0.], [0., 0., 0., 1.]])
```

Comment:
Thanks!  Is counterclockwise rotation more common? Or you can specify the direction of rotation in the comments.

Reply:
Yes, of couse all the rotations is based on counterclockwise. I think this is a standard which should be stated in the official documentation like other math library did.

I will note this in pr conversation for now, and later note it on the Matrix.md